File: blue_compare.rb
class BlueCompare
  attr_reader :award

  def initialize(award)
    @award = award
  end

  def update
    award.expires_in -= 1
    return if award.quality >= 50
    return award.quality = 0 if award.expires_in.negative?

    award.quality += 1
    award.quality += 1 if award.expires_in < 10
    award.quality += 1 if award.expires_in < 5
  end
end

File: blue_compare.py
from award import Award

class BlueCompare(Award):
	def __init__(self, expires_in=None, quality=None):
		super(BlueCompare, self).__init__('Blue Compare', expires_in, quality)

	def update_quality(self):
		if self.quality < 50:
			self.quality += 1
			if self.expires_in < 11 and self.quality < 50:
				self.quality += 1
			if self.expires_in < 6 and self.quality < 50:
				self.quality += 1 


		self.expires_in -= 1
		if self.expires_in < 0:
			self.quality = 0